File: android/shortsblockerplugin/BootReceiver.java

package app.lovable.shortsblockerplugin;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.os.Build;
import android.provider.Settings;

public class BootReceiver extends BroadcastReceiver {
    @Override
    public void onReceive(Context context, Intent intent) {
        if (Intent.ACTION_BOOT_COMPLETED.equals(intent.getAction())) {
            // Check if accessibility service is enabled
            String enabledServices = Settings.Secure.getString(
                    context.getContentResolver(),
                    Settings.Secure.ENABLED_ACCESSIBILITY_SERVICES);
            
            if (enabledServices != null && enabledServices.contains(context.getPackageName() + "/app.lovable.shortsblockerplugin.AccessibilityServicePlugin$ShortsBlockerService")) {
                // Start our service
                Intent serviceIntent = new Intent(context, AccessibilityServicePlugin.ShortsBlockerService.class);
                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
                    context.startForegroundService(serviceIntent);
                } else {
                    context.startService(serviceIntent);
                }
            }
        }
    }
}

File: android/src/main/java/app/lovable/plugin/BootReceiver.kt

package app.lovable.plugin

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.os.Build
import android.provider.Settings

class BootReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        if (Intent.ACTION_BOOT_COMPLETED == intent.action) {
            // Check if accessibility service is enabled
            val enabledServices = Settings.Secure.getString(
                context.contentResolver,
                Settings.Secure.ENABLED_ACCESSIBILITY_SERVICES
            )
            
            if (enabledServices?.contains(
                context.packageName + "/app.lovable.plugin.ShortsBlockerPlugin\$YoutubeShortsBlockerService"
            ) == true) {
                // Start our service
                val serviceIntent = Intent(context, ShortsBlockerPlugin.YoutubeShortsBlockerService::class.java)
                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
                    context.startForegroundService(serviceIntent)
                } else {
                    context.startService(serviceIntent)
                }
            }
        }
    }
}
